assembly-ide/src/assembly/bin/che.sh

```diff
@@ -10,20 +10,436 @@
+# If exit=1 then output error message + usage
+# if exit=0 then print usage
 
-if [ ! -d "${ASSEMBLY_BIN_DIR}" ]; then
-  echo "$(tput setaf 1)Could not find the app server binaries in Eclipse Che."$(tput sgr0)
-  exit 1
+WRONG="
+
+Looks like something went wrong. Possible issues: 
+  1. (Win | Mac) VirtualBox not installed           ==> Rerun Docker Toolbox installation
+  2. (Win | Mac) Docker Machine not installed       ==> Rerun Docker Toolbox installation
+  3. (Win | Mac) Docker is not reachable            ==> Docker VM failed to start
+  4. (Win | Mac) Docker ok, but docker ps fails     ==> Docker environment variables not set properly
+  5. (Linux) Docker is not reachable                ==> Is Docker installed ==> wget -qO- https://get.docker.com/ | sh
+  6. Could not find the Che app server              ==> Did /tomcat get moved away from CHE_HOME?
+  7. Did you use the right parameter syntax?        ==> See usage
+
+We have seen issues with VirtualBox on windows where your VM gets corrupted when your computer is suspended while the VM is still running. This will appear as SSH or ethernet connection issues. This is rare, but if encountered, current known solution is to uninstall VirtualBox and Docker Toolbox, and then reinstall."
+
+CHE_VARIABLES="
+
+Che Environment Variables:
+  (REQUIRED) JAVA_HOME                              ==> Location of Java runtime
+  (REQUIRED: WIN|MAC) DOCKER_TOOLBOX_INSTALL_PATH   ==> Location of Docker Toolbox
+  (REQUIRED: WIN|MAC) VBOX_MSI_INSTALL_PATH         ==> Location of VirtualBox  
+  (OPTIONAL) CHE_HOME                               ==> Directory where Che is installed
+  (OPTIONAL) CHE_LOCAL_CONF_DIR                     ==> Directory with custom Che .properties files
+  (OPTIONAL) CHE_LOGS_DIR                           ==> Directory for Che output logs"
+
+USAGE="
+
+Usage: 
+  che [-i] [-i:tag] [-p:port] [run | start | stop]
+
+     -i,      --image        Launches Che within a Docker container using latest image
+     -i:tag,  --image:tag    Launches Che within a Docker container using specific image tag
+     -p:port, --port:port    Port that Che server will use for HTTP requests; default=8080
+     -h,      --help         Show this help
+     run                     Starts Che application server in current console
+     start                   Starts Che application server in new console
+     stop                    Stops Che application server"
+
+function usage {
+  echo "$USAGE"
+}
+
+function error_exit {
+  echo "
+$1"
+  echo "$WRONG $CHE_VARIABLES $USAGE"
+  JUMP_TO_END=true
+}
+
+# Run the finish function if exit signal initiated
+trap exit SIGHUP SIGINT SIGTERM
+
+function parse_command_line {
+
+  ### Parse command line parameters
+  USE_DOCKER=false
+  USE_DOCKER_TAG=latest
+  USE_PORT=8080
+  USE_HELP=false
+  USE_SERVER_ACTION=run
+
+  for command_line_option in "$@"
+  do
+  case $command_line_option in
+    -i|--image)
+      USE_DOCKER=true
+      shift # past argument
```

Comment:
I don't see direct usage of arguments here (except "$@") so it looks like **shift** is useless in this function